src/characters.rs

```diff
@@ -0,0 +1,59 @@
+use crate::AppState;
+use actix_web::{get, web, HttpRequest, HttpResponse, Responder};
+use serde::Serialize;
+use std::collections::HashMap;
+
+#[derive(Debug, Serialize)]
+pub struct Character {
+    name: String,
+    line_count: i32,
+    speaks_count: i32,
+    id: String,
+    actor_id: Option<i32>,
+}
+
+#[get("/characters/{script_id}")]
+pub async fn get_characters(
+    data: web::Data<AppState>,
+    script_id: web::Path<i32>,
+    req: HttpRequest,
+) -> impl Responder {
+    // Gets the optional query string out of the request and grabs the actor_id as an int
+    let query = req.query_string();
+    let actor_id: i32;
+    if query != "" {
+        actor_id = query.split('=').last().unwrap().parse::<i32>().unwrap();
```

Comment:
Now that you've discovered that the trailing `/` in the js fetch is what was causing this endpoint not to work, can you see if using a Query extractor will work here?  
Seem like a struct with `Option` fields _should_ work.  
https://stackoverflow.com/questions/54406029/how-can-i-parse-query-strings-in-actix-web  
  
That would be more concise/safer than string manipulation.

Reply:
Also, I think a more idiomatic way to assign actor_id in rust is to assign the result of the `if` expression:  
```rust
let actor_id = if query != "" {
    actor_id = query.split('=').last().unwrap().parse::<i32>().unwrap()
} else {
    actor_id = -1
}
```

Reply:
Works like a charm!